HIP/hip-867.md

```diff
@@ -0,0 +1,119 @@
+status: Draft
+created: 2024-01-24
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/872
+---
+
+## Abstract
+
+Add the Ethereum KZG Point Evaluation Precompile from EIP-4844: Shard Blob
+Transactions.
+
+## Motivation
+
+One part of [EIP-4844](https://eips.ethereum.org/EIPS/eip-4844) is the KZG Point
+Evaluation function, a cryptographic precompile that can be used with zero
+knowledge proofs as well as proving that certain subsections of a blob are
+present without introducing the entire blob. This precompile adds value to the
+other parts of EIP-4844, but it does not require them. Hedera will expose
+support for this precompile without introducing other parts of the EIP.
+
+## Rationale
+
+While Hedera will not be supporting most EIP-4844 features the KZG Point
+Evaluation Precompile is cleanly severable from the other integrated features.
+This precompile, when accompanied by the associated trusted setup, can be viewed
+as a cryptographic primitive similar to ECDSA. The ability to prove subsets of
+hashed data provides unique functionality not found in other cryptographic
+precompiles. Ethereum apps may use the KZG features without any interation with
+blob varying transactions, and it makes sense to allow Hedera Smart Contract
+apps the same latitude.
+
+This precompile will be brought in with the same semantics and calling
+conventions as Ethereum Mainnet. It also requires a customizable data file known
+as the trusted setup. For maximum compatibility the same trusted setup from
+Ethereum Mainnet will be used.
+
+## User stories
+
+* As a smart contract developer I want to use the EIP-4844 Point Evaluation
+  Precompile in my smart contract applications.
+
+## Specification
+
+There are no changes needed to Hedera Services and the JSON-RPC relay.
+
+### EVM Implementation
+
+The [KZG Point Evaluation precompile](https://eips.ethereum.org/EIPS/eip-4844#point-evaluation-precompile)
+will be implemented as it is in EIP-4844.
+
+The address for the precompile will be the same as Ethereum
+Mainnet (`0x000000000000000000000000000000000000000A`).
+
+The point evaluation will use the
+same [trusted setup](https://github.com/ethereum/c-kzg-4844/blob/main/src/trusted_setup.txt)
+as Ethereum Mainnet.
+
+This is the same out-of-the-box setup provided by the Besu EVM library, there
+should be little to zero Hedera code needed to enable this.
+
+## Backwards Compatibility
+
+The core EVM library shipping with Hedera as of version 0.46 already contains
+the needed EVM support. The activation will add a new Hedera EVM version that
+will activate all the Cancun support in one release. The precompile should
+automatically be part of the Besu EVM Library implementation.
+
+## Security Implications
+
+It is expected the gas cost of the precompile will fall in line with compute
+usage during the execution. Performance tests will need to be executed to
+validate that heavy use of the precompile matches this expectation.
```

Comment:
I don't actually see in EIP-4844 where the gas cost of the KZG precompiles (i.e. the "point evaluation precompile) is specified (only the gas cost for blob transactions is in there).  Where is it?  (Would help too if it was added to the references, as it is mentioned here as a possible gotcha to be checked.)

Reply:
https://eips.ethereum.org/EIPS/eip-4844#point-evaluation-precompile using the constant `POINT_EVALUATION_PRECOMPILE_GAS` defined in [the Parameters](https://eips.ethereum.org/EIPS/eip-4844#parameters)  All in the 4844 EIP doc.